cdap-master/src/main/java/co/cask/cdap/data/runtime/main/MasterServiceMain.java

```diff
@@ -182,12 +183,20 @@ public void start() {
       LOG.error("Could not disable caching of URLJarFiles. This may lead to 'too many open files` exception.", e);
     }
 
-    ConfigurationLogger.logImportantConfig(cConf);
-
-    LOG.info("Client Hadoop version: {}", ClientVersions.getHadoopVersion());
-    LOG.info("Client ZooKeeper version: {}", ClientVersions.getZooKeeperVersion());
-    LOG.info("Client Kafka version: {}", ClientVersions.getKafkaVersion());
-    LOG.info("Client Hive version: {}", ExploreServiceUtils.getHiveVersion());
+    LOG.info("Hadoop subsystem versions:");
+    LOG.info("  Hadoop version: {}", ClientVersions.getHadoopVersion());
+    LOG.info("  ZooKeeper version: {}", ClientVersions.getZooKeeperVersion());
+    LOG.info("  Kafka version: {}", ClientVersions.getKafkaVersion());
+    if (cConf.getBoolean(Constants.Explore.EXPLORE_ENABLED)) {
+      LOG.info("Client Hive version: {}", ExploreServiceUtils.getHiveVersion());
+    }
+    if (!startupChecker.runSanityChecks()) {
+      LOG.error("Errors while starting up master. " +
+                  "Please check the logs, address all errors, then start the master service again. " +
+                  "Startup checks can also be disabled by setting {} to false in cdap-site.xml",
```

Comment:
An option as in a command line option instead of a config setting?  A lot of users will be starting the master through cloudera manager or ambari, so it's not easier for those users. Then once there are options, how do you choose what should be an option versus a config setting? 

I can remove this line so we're not tied to anything though.